enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -373,7 +376,9 @@ type Subnet struct {
     // +optional
     // +listType=atomic
     // +kubebuilder:validation:XValidation:rule=`self.all(x, self.exists_one(y, x == y))`,message="roles cannot contain duplicates"
-    // +kubebuilder:validation:MaxLength=4
+    // +kubebuilder:validation:XValidation:rule=`!self.exists(r, r.type == 'EdgeNode') || self.size() == 1`,message="EdgeNode roles cannot be combined with any other roles"
+    // +kubebuilder:validation:XValidation:rule=`!(self.exists(r1, r1.type == 'ControlPlaneExternalLB') && self.exists(r2, r2.type == 'ControlPlaneInternalLB'))`,message="roles cannot contain both ControlPlaneExternalLB and ControlPlaneInternalLB"
```

Comment:
An alternative to this validation, in case you find this one easier to reason about
```
self.filter(r, r.type == 'ControlPlaneExternalLB' || r.type == 'ControlPlaneInternalLB').size() < 2
```